backend/filen/filen.go

```diff
@@ -0,0 +1,1084 @@
+	return f.Put(ctx, in, src, options...)
+}
+
+// Mkdir makes the directory (container, bucket)
+//
+// Shouldn't return an error if it already exists
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	dirObj, err := f.filen.FindDirectoryOrCreate(ctx, f.resolvePath(f.Enc.FromStandardPath(dir)))
+	if err != nil {
+		return err
+	}
+	if dir == f.root.path {
+		f.root.directory = dirObj
+	}
+	return nil
+}
+
+// Rmdir removes the directory (container, bucket) if empty
+//
+// Return an error if it doesn't exist or isn't empty
+func (f *Fs) Rmdir(ctx context.Context, dir string) error {
+	// find directory
+	resolvedPath := f.resolvePath(f.Enc.FromStandardPath(dir))
+	//if resolvedPath == f.root.path {
+	//	return fs.ErrorDirNotFound
+	//}
+	directory, err := f.filen.FindDirectory(ctx, resolvedPath)
+	if err != nil {
+		return err
+	}
+	if directory == nil {
+		return errors.New("directory not found")
+	}
+
+	files, dirs, err := f.filen.ReadDirectory(ctx, directory)
+	if err != nil {
+		return err
+	}
+	if len(files) > 0 || len(dirs) > 0 {
+		return errors.New("directory is not empty")
+	}
+
+	// trash directory
+	err = f.filen.TrashDirectory(ctx, directory)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Directory is Filen's directory type
+type Directory struct {
+	fs        *Fs
+	path      string
+	directory types.DirectoryInterface
+}
+
+// Fs returns read only access to the Fs that this object is part of
+func (dir *Directory) Fs() fs.Info {
+	return dir.fs
+}
+
+// String returns a description of the Object
+func (dir *Directory) String() string {
+	if dir == nil {
+		return "<nil>"
+	}
+	return dir.Remote()
+}
+
+// Remote returns the remote path
+func (dir *Directory) Remote() string {
+	return dir.fs.Enc.ToStandardPath(dir.path)
+}
+
+// ModTime returns the modification date of the file
+// It should return a best guess if one isn't available
+func (dir *Directory) ModTime(ctx context.Context) time.Time {
+	directory, ok := dir.directory.(*types.Directory)
+	if !ok {
+		return time.Time{} // todo add account creation time?
+	}
+
+	if directory.Created.IsZero() {
+		obj, err := dir.fs.filen.FindDirectory(ctx, dir.fs.resolvePath(dir.path))
+		newDir, ok := obj.(*types.Directory)
+		if err != nil || !ok {
+			return time.Now()
+		}
+		directory = newDir
+		dir.directory = newDir
+	}
+	return directory.Created
+}
+
+// Size returns the size of the file
+//
+// filen doesn't have an efficient way to find the size of a directory
+func (dir *Directory) Size() int64 {
+	return -1
+}
+
+// Items returns the count of items in this directory or this
+// directory and subdirectories if known, -1 for unknown
+func (dir *Directory) Items() int64 {
+	return -1
+}
+
+// ID returns the internal ID of this directory if known, or
+// "" otherwise
+func (dir *Directory) ID() string {
+	return dir.directory.GetUUID()
+}
+
+// File is Filen's normal file
+type File struct {
```

Comment:
Could you call this `Object` please to match the other backends otherwise it is going to confuse us!

The methods should be `func (o *Object) Method()` also.